nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Run/PullRequestTextGenerator.cs

```diff
@@ -0,0 +1,44 @@
+using NuGet.Versioning;
+
+using NuGetUpdater.Core.Run.ApiModel;
+
+namespace NuGetUpdater.Core.Run;
+
+public class PullRequestTextGenerator
+{
+    public static string GetPullRequestTitle(Job job, ReportedDependency[] updatedDependencies, DependencyFile[] updatedFiles, string? dependencyGroupName = null)
+    {
+        // simple version looks like
+        //   Update Some.Package to 1.2.3
+        // if multiple packages are updated to multiple versions, result looks like:
+        //   Update Package.A to 1.0.0, 2.0.0; Package.B to 3.0.0, 4.0.0
+        var dependencySets = updatedDependencies
+            .GroupBy(d => d.Name, StringComparer.OrdinalIgnoreCase)
+            .OrderBy(g => g.Key, StringComparer.OrdinalIgnoreCase)
+            .Select(g => new
+            {
+                Name = g.Key,
+                Versions = g
+                    .Where(d => d.Version is not null)
+                    .Select(d => d.Version!)
+                    .OrderBy(d => NuGetVersion.Parse(d))
+                    .ToArray()
```

Comment:
q: Is there a reason we need to force evaluation with `.ToArray()` here or is it just habit?

Reply:
It's purely so I can see the result in the debugger without expanding the "Results" element.  Similar reason I set "var title = ..." then immediately "return title;"; it's so I have handy breakpoint locations and can immediately see the value.